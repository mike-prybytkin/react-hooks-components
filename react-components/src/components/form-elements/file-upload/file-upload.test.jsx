import React from 'react';
import { render } from '@testing-library/react';
import mockText from '../../../mocks/text';
import FileUpload from './file-upload';

const setUp = () => {
  return render(
    <FileUpload labelType={mockText.labelFileUpload} fileUploadRef={undefined} id={'test'} />
  );
};

describe('FileUpload component', () => {
  it('should correctly render date input with a className', () => {
    const { container } = setUp();
    expect(container.getElementsByClassName('file-upload').length).toBe(1);
  });

  it('should create snapshot', () => {
    const { asFragment } = setUp();
    expect(asFragment()).toMatchSnapshot();
  });
});
