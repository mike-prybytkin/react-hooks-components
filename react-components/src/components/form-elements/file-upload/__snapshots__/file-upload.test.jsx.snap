// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUpload component should create snapshot 1`] = `
<DocumentFragment>
  <div
    class="file-upload"
  >
    <label
      class="file-upload__label"
    >
      <input
        accept="image/*"
        class="file-upload__input"
        id="test"
        name="test"
        type="file"
      />
      Add your photo
    </label>
  </div>
</DocumentFragment>
`;
